modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/SharingUrlTimestampParser.kt

```diff
@@ -0,0 +1,37 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+class SharingUrlTimestampParser {
+    fun parseTimestamp(t: String): Pair<Duration?, Duration?> {
+        val intervalPattern = Regex("""(\d*\.?\d*)?,?(\d*\.?\d*)?""") // t=10,20 or t=,20 or t=10
+        val hmsPattern = Regex("""(?:(\d+)h)?(?:(\d+)m)?(?:(\d+(?:\.\d+)?)s)?""") // t=10h20m30s or t=10m30s or or 10m or t=10s
+        val hhmmssFractionPattern = Regex("""(\d+):(\d{2}):(\d{2})(?:\.(\d+))?""") // t=10:20:30.456
+
+        hmsPattern.matchEntire(t)?.let { match ->
+            val hours = match.groups[1]?.value?.toInt() ?: 0
+            val minutes = match.groups[2]?.value?.toInt() ?: 0
+            val seconds = match.groups[3]?.value?.toDouble() ?: 0.0
+            val totalSeconds = ((hours * 3600) + (minutes * 60) + seconds)
+            return Pair(totalSeconds.seconds, null)
+        }
+
+        hhmmssFractionPattern.matchEntire(t)?.let { match ->
+            val hours = match.groups[1]?.value?.toLong() ?: 0
+            val minutes = match.groups[2]?.value?.toLong() ?: 0
+            val seconds = match.groups[3]?.value?.toLong() ?: 0
+            val fraction = match.groups[4]?.value?.toDouble() ?: 0.0
+            val totalSeconds = (hours * 3600) + (minutes * 60) + seconds + (fraction / 1000.0)
```

Comment:
The `fraction / 1000.0` part is faulty. For example this test fails and evaluates timestamp to `5d 18h 53m 21s`.

```kotlin
assertEquals(Pair(1500.milliseconds, null), parser.parseTimestamp("00:00:01.500000000"))
```